### PR TITLE
fix: extension build workflow

### PR DESCRIPTION
**Pull request by alter-eggo**
> Try out Leather build 14a4dc7 — [Extension build](https://github.com/leather-wallet/extension/actions/runs/9249993782), [Test report](https://leather-wallet.github.io/playwright-reports/fix-build-yml), [Storybook](https://fix-build-yml--65982789c7e2278518f189e7.chromatic.com), [Chromatic](https://www.chromatic.com/library?appId=65982789c7e2278518f189e7&branch=fix-build-yml)

This pr fixes extension build workflow to ensure usage of sanitised branch names, fixes https://github.com/leather-wallet/issues/issues/72

## Summary by CodeRabbit

- **Chores**
  - Improved the build script to sanitize branch names for better compatibility.
  - Updated the process for parsing branch names for Storybook to use the sanitized branch name directly.

### CODE REVIEW

**Comment by markmhendrickson** [comment]
Is there any issue to link here?

**Comment by alter-eggo** [comment]
@markmhendrickson it's fix for h1 report, should we create sep issues for them?

**Review by kyranjamie — APPROVED** [review]
This LGTM 👍🏻

File: .github/workflows/build-extension.yml [hunk 1]
```diff
@@ -72,19 +72,21 @@ jobs:
       - build
     steps:
       - name: Extract branch name
-        run: echo "branch=${GITHUB_HEAD_REF:-${GITHUB_REF#refs/heads/}}" >> $GITHUB_OUTPUT
         id: extract_branch
+        run: |
+          echo "branch=${GITHUB_HEAD_REF:-${GITHUB_REF#refs/heads/}}" >> $GITHUB_ENV
+          echo "BRANCH_NAME=$(echo ${GITHUB_HEAD_REF:-${GITHUB_REF#refs/heads/}} | sed 's/[^a-zA-Z0-9_-]//g')" >> $GITHUB_ENV
```
Comment (kyranjamie) [inline .github/workflows/build-extension.yml]: Aren't we doing the same thing twice here? branch and BRANCH_NAME are the same?

Comment (alter-eggo) [reply .github/workflows/build-extension.yml]: 🤦‍♂️ yeah, thanks

**Review by kyranjamie — CHANGES_REQUESTED** [review]
Doesn't work

```
> Try out Leather build 09ded0a — [Extension build](https://github.com/leather-wallet/extension/actions/runs/9222912951), [Test report](https://leather-wallet.github.io/playwright-reports/${BRANCH_NAME}), [Storybook](https://${BRANCH_NAME}--65982789c7e2278518f189e7.chromatic.com), [Chromatic](https://www.chromatic.com/library?appId=65982789c7e2278518f189e7&branch=${BRANCH_NAME})<!-- Sticky Header Marker -->
```

**Comment by kyranjamie** [comment]
Another idea might be to entirely prevent these from running on forks?

File: .github/workflows/build-extension.yml [hunk 2]
```diff
@@ -72,19 +72,20 @@ jobs:
...
-        run: echo "STORYBOOK_BRANCH=`echo ${{ steps.extract_branch.outputs.branch }} | sed 's/\//\-/g'`" >> $GITHUB_OUTPUT
+        run: echo "STORYBOOK_BRANCH=${BRANCH_NAME}" >> $GITHUB_ENV
 
       - uses: kyranjamie/pull-request-fixed-header@v1.0.1
         env:
           EXTENSION_BUILD_LINK: '[Extension build](https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }})'
-          TEST_REPORT_LINK: '[Test report](https://leather-wallet.github.io/playwright-reports/${{ steps.extract_branch.outputs.branch }})'
-          STORYBOOK_LINK: '[Storybook](https://${{ steps.storybook_branch.outputs.STORYBOOK_BRANCH }}--65982789c7e2278518f189e7.chromatic.com)'
-          CHROMATIC_LINK: '[Chromatic](https://www.chromatic.com/library?appId=65982789c7e2278518f189e7&branch=${{ steps.extract_branch.outputs.branch }})'
+          TEST_REPORT_LINK: '[Test report](https://leather-wallet.github.io/playwright-reports/${BRANCH_NAME})'
```
Comment (kyranjamie) [inline .github/workflows/build-extension.yml]: `env.BRANCH_NAME`?

File: .github/workflows/build-extension.yml [hunk 3]
```diff
@@ -65,26 +65,28 @@ jobs:
           path: dist
 
   set-download-link:
-    if: github.repository == 'leather-wallet/extension' && github.actor != 'dependabot[bot]'
+    if: github.event.pull_request.head.repo.fork == false && github.repository == 'leather-wallet/extension' && github.actor != 'dependabot[bot]'
```
Comment (edgarkhanzadian) [inline .github/workflows/build-extension.yml]: I think `github.event.pull_request.head.repo.fork == false` is redundant if we are going to check for `github.repository == 'leather-wallet/extension'` anyway. Forking is going to change that name to `<username>/extension`.

**Review by edgarkhanzadian — APPROVED** [review]
Great work!

File: .github/workflows/build-extension.yml (hunk 3 above)
Comment (kyranjamie) [reply .github/workflows/build-extension.yml]: +1. If we want to disable the job entirely, we need to move this condition (which is okay as it is) to the root level

Comment (alter-eggo) [reply .github/workflows/build-extension.yml]: yeah, forgotten to remove

**Review by kyranjamie — APPROVED** [review]

**Pull request merged**